### DISCUSSION TITLE
On demand Server-side fetch request

### DISCUSSION THREAD

**Discussion by Ashesh3** [vikejs/vike#996] [Discussion] [answered]
Hey, in my `index.page.tsx`, I want to make a `fetch` request to a URL but I want this to be done server side (and not on the user's browser). How do I achieve this? I have a button that needs to make a fetch request (but the server needs to make that http request). My thoughts are that I would need to add a custom expressJS endpoint in `server/index.ts` that would receive requests from my browser and then make the fetch call server side then return the response. Is there a better way? From the docs I only see a way to fetch data before render, and not what I am looking for.

**Comment by brillout** [comment, marked as answer]
See https://vite-plugin-ssr.com/api-routes. (Thanks for reaching out, I'm realizing the docs's structure can be improved.)
